testbed/tests/app/test_desktop.py

```diff
@@ -348,50 +348,84 @@ async def test_show_hide_cursor(app, app_probe):
 
-        # When all windows are hidden, WinForms and Cocoa return None, while GTK
-        # returns the last active window.
         main_window.hide()
-        assert app.current_window in [None, main_window]
+        assert app.current_window is None
 
         main_window.show()
         assert app.current_window == main_window
     finally:
         main_window.show()
 
-    window1 = toga.Window("Test Window 1", position=(150, 150), size=(200, 200))
-    window2 = toga.Window("Test Window 2", position=(400, 150), size=(200, 200))
-    window3 = toga.Window("Test Window 3", position=(300, 400), size=(200, 200))
+    window1 = toga.Window(title="Test Window 1", position=(150, 150), size=(200, 200))
+    window2 = toga.Window(title="Test Window 2", position=(400, 150), size=(200, 200))
+    window3 = toga.Window(title="Test Window 3", position=(300, 400), size=(200, 200))
 
     window1.content = toga.Box(style=Pack(background_color=REBECCAPURPLE))
     window2.content = toga.Box(style=Pack(background_color=CORNFLOWERBLUE))
     window3.content = toga.Box(style=Pack(background_color=FIREBRICK))
 
-    # We don't need to probe anything window specific; we just need
-    # a window probe to enforce appropriate delays.
-    window1_probe = window_probe(app, window1)
-
     window1.show()
     window2.show()
     window3.show()
 
-    await window1_probe.wait_for_window("Extra windows added")
+    await main_window_probe.wait_for_window("Extra windows added")
 
+    # When a window without any dialog is made the current_window,
+    # then `app.current_window` should return the specified window.
     app.current_window = window2
-    await window1_probe.wait_for_window("Window 2 is current")
+    await main_window_probe.wait_for_window("Window 2 is current")
     if app_probe.supports_current_window_assignment:
         assert app.current_window == window2
 
     app.current_window = window3
-    await window1_probe.wait_for_window("Window 3 is current")
+    await main_window_probe.wait_for_window("Window 3 is current")
     if app_probe.supports_current_window_assignment:
         assert app.current_window == window3
 
+    # When a dialog is in focus, app.current_window should return the
+    # previously active window.
+    def test_current_window_in_presence_of_dialog(dialog, previous_active_window):
+        # The public API should report that the previous window is the current window
+        if app_probe.supports_current_window_assignment:
+            assert app.current_window == previous_active_window
+
+        # But, the backend should be reporting that the dialog is in focus
+        app_probe.assert_dialog_in_focus(dialog)
+
+    # Test in presence of window modal dialog
+    previous_active_window = app.current_window
+    window_modal_info_dialog = toga.InfoDialog("Window Modal Dialog Info", "Some info")
+    main_window_probe.setup_info_dialog_result(
+        window_modal_info_dialog,
+        pre_close_test_method=partial(
```

Comment:
Given that the dialog is available at the point the test method is being invoked (and would be of general utility to *any* dialog test method), and the previous_active_window is a constant - why not make `dialog` a required first argument of the test method. and make the usage of `previous_active_window` a closure? (i.e., the test method referencing a variable that is outside the context of the method).